src/aggregator/aggregator/follower_flush_mgr.go

```diff
@@ -279,12 +307,39 @@ func (mgr *followerFlushManager) CanLead() bool {
 	return true
 }
 
+// canLeadNotFlushed determines whether the follower can takeover leadership
+// for the window that was not (yet) flushed by leader.
+// This is case is possible when leader encounters a metric at some resolution
+// window for the first time in the shard it owns.
+func (mgr *followerFlushManager) canLeadNotFushed(windowSize time.Duration) bool {
+	earliestPossibleWindowStart := mgr.nowFn().Add(-windowSize)
```

Comment:
Oh btw, let's use a consistent `now` for this--we can call `nowFn` at the top and then pass it through.

Reply:
Well, it would make code chattier, because `mgr.nowFn()` is not used in the callsites of `canLeadNotFlushed()`. But `mgr.nowFn()` is consistently used in other methods not related to `canLead`.

Reply:
Yeah not a lot for it I think; I think it's still worth getting a consistent snapshot if we can (obviously only need to pass it through where we need it).

Reply:
Oh got it. Done, thanks!